HEC.FDA.Model/paireddata/PairedData.cs

```diff
@@ -366,9 +367,30 @@ public void ForceMonotonicity(double max = double.MaxValue, double min = double.
             }
             Yvals = update;
         }
-        public void ForceStrictMonotonicity(double max = double.MaxValue, double min = double.MinValue)
+        public void ForceStrictMonotonicityTopDown(double max = double.MaxValue, double min = double.MinValue)
+        {
+            double[] update = new double[Yvals.Length];
```

Comment:
This is low enough in the stack, (Litteral Billions of paired data are created for a given compute). I want to make sure we're being real memory stingy. I think we could do this in place rather than allocating a new array and then copying it back over.  

If that doesn't make immediate sense, lets talk about it in the office.